uk.co.compucorp.civicrm.hrcore/hrcore.php

```diff
@@ -134,3 +134,23 @@ function hrcore_civicrm_apiWrappers(&$wrappers, $apiRequest) {
     $wrappers[] = new CRM_HRCore_APIWrapper_DefaultLimitRemover();
   }
 }
+
+/**
+ * Implementation of hook_civicrm_pre hook.
+ *
+ * @param string $op
+ * @param string $objectName
+ * @param int $objectId
+ * @param array $params
+ */
+function hrcore_civicrm_pre($op, $objectName, $objectId, &$params) {
+  if ($objectName === 'Address' && ($op === 'edit' || $op === 'create')) {
+    $homeLocation = civicrm_api3('LocationType', 'getsingle', [
+      'name' => 'Home',
+    ]);
+
+    if ($params['location_type_id'] == $homeLocation['id']) {
+      $params['is_primary'] = 1;
+    }
+  }
```

Comment:
We should avoid adding business logic directly into the hook.  One approach we've been trying to follow in other extensions is make make the hooks work like event dispatchers. It would like this:

```php
function hrcore_civicrm_pre($op, $objectName, $objectId, &$params) {
  $listeners = [
     new Listener1(),
     new Listener2(),
     new Listerner3(),
     ...
  ];

  foreach($listeners as $listener) {
    $listener->handle($op, $objectName, $objectId, &$params);
  }
}
```

As for the listener, you could have something like:
```php

class CRM_HRCore_Hook_Pre_PrimaryAddressSetter {
  public function handle($op, $objectName, $objectId, &$params) {
    if(!$this->shouldHandle($op, $objectName)) {
       return;
    }
    
    $homeLocation = civicrm_api3('LocationType', 'getsingle', [
      'name' => 'Home',
    ]);

    if ($params['location_type_id'] == $homeLocation['id']) {
      $params['is_primary'] = 1;
    }
  
  }

  private function shouldHandle() {
    // some logic here
  }
}

```

I know this might look like too much at this point, but it's a measure to prevent hooks to become a mess of if and else, like what we have in some of the other extensions